linode/instance/resource.go

```diff
@@ -818,6 +825,17 @@ func updateResource(ctx context.Context, d *schema.ResourceData, meta interface{
 		}
 	}
 
+	if d.HasChange("ipv4") {
+		fmt.Println("Change detected")
+		oldIPv4, newIPv4 := d.GetChange("ipv4")
+		oldSet := oldIPv4.(*schema.Set)
+		newSet := newIPv4.(*schema.Set)
+		if len(newSet.Difference(oldSet).List()) > 0 {
+			// Log a warning that direct IP changes are not supported
+			return diag.Errorf("Changing IPv4 addresses after instance creation is not directly supported. The attempted change will be ignored.")
+		}
+	}
```

Comment:
It looks like the ipv4 addresses can't be updated, right? If so, I think we can simplify this part to be just returning the error when change detected: 
```suggestion
	if d.HasChange("ipv4") {
		// Log a warning that direct IP changes are not supported
		return diag.Errorf("Changing IPv4 addresses after instance creation is not directly supported. The attempted change will be ignored.")
	}
```

Reply:
Yeah I have removed the part where I am storing the diff